desktop/core/src/desktop/js/reactComponents/FileChooser/PathBrowser/PathBrowser.test.tsx

```diff
@@ -0,0 +1,175 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import { render, screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+
+import PathBrowser from './PathBrowser';
+import { debug } from 'webpack';
+
+const handleFilePathChangeMock = jest.fn();
+
+const breadcrumbsTestConfig1 = [
+  {
+    url: 'abfs://',
+    label: 'abfs://'
+  },
+  {
+    url: 'abfs://test',
+    label: 'test'
+  },
+  {
+    url: 'abfs://test/test1',
+    label: 'test1'
+  }
+];
+
+const breadcrumbsTestConfig2 = [
+  {
+    url: 'abfs://',
+    label: 'abfs://'
+  },
+  {
+    url: 'abfs://test',
+    label: 'test'
+  },
+  {
+    url: 'abfs://test/test1',
+    label: 'test1'
+  },
+  {
+    url: 'abfs://test/test1/test2',
+    label: 'test2'
+  },
+  {
+    url: 'abfs://test/test1/test2/a very very very long test label',
+    label: 'a very very very long test label'
+  },
+  {
+    url: 'abfs://test/test1/test2/a very very very long test label/a very very very long test label 1',
+    label: 'a very very very long test label 1'
+  },
+  {
+    url: 'abfs://test/test1/test2/a very very very long test label/a very very very long test label 1/a very very very long test label 2',
+    label: 'a very very very long test label 2'
+  }
+];
+
+describe('Pathbrowser', () => {
+  describe('Pathbrowser breadcrumbs', () => {
+    test('renders the specified seperator to seperate the breadcrumbs', () => {
+      render(
+        <PathBrowser
+          handleFilePathChange={handleFilePathChangeMock}
+          breadcrumbs={breadcrumbsTestConfig1}
+          seperator={'%'}
+          showIcon
+        />
+      );
+      const seperator = screen.getAllByTestId('hue-path-browser-breadcrumb-seperator');
+      expect(within(seperator[0]).getByText('%')).toBeVisible();
+    });
+
+    test('renders breadcrumbs without dropdown if there are less than or equal to 3 breadcrumbs', () => {
+      const rendered = render(
+        <PathBrowser
+          handleFilePathChange={handleFilePathChangeMock}
+          breadcrumbs={breadcrumbsTestConfig1}
+          seperator={'/'}
+          showIcon
+        />
+      );
+      expect(rendered.queryByTestId('hue-path-browser-dropdown-btn')).toBeNull();
+    });
+
+    test('renders breadcrumbs with dropdown if there are more than 3 breadcrumbs', () => {
+      const rendered = render(
+        <PathBrowser
+          handleFilePathChange={handleFilePathChangeMock}
+          breadcrumbs={breadcrumbsTestConfig2}
+          seperator={'/'}
+          showIcon
+        />
+      );
+      expect(rendered.queryByTestId('hue-path-browser-dropdown-btn')).toBeVisible();
```

Comment:
You are actually not testing that the breadcrumbs are rendered with a dropdown, only that there is a button available. A better test is to also click the button and verify that the breadcrumbs are there.

Reply:
There was an issue with queryAllByTestId hence I had removed that particular test. I have since added that in the recent commit.